Fix duplicate volume warnings and missing material IDs

sethrj opened the pull request:
I think this fixes the `other.material_id < params_.materials.size` issue reported by @whokion in #750. The material IDs for reflected volumes were not being captured because the mangled volume names were not matching (see https://github.com/celeritas-project/celeritas/pull/680#issuecomment-1541946789 ).

This also fixes a bunch of warnings about duplicate volume names any time a boolean is present in the geometry: vecgeom was generating temporary unplaced volumes which show up multiple times in the "registry" of volumes (with a unique counter) but are only used as part of the underlying implementation and never encountered during navigation.

sethrj committed:
Remove labels from temporary bool logical volumes

sethrj committed:
Do not mangle based on underlying reflected volume

whokion commented:
To verify the fix for the material id check, should we support a GPU version of the Reflected Solid first?  Currently, there is no way to confirm the fix as the first assertion happens earlier with unsupported Reflected volumes (i.e, at [GenericSolid](https://github.com/sethrj/celeritas/blob/cms-conversion-fixes/src/celeritas/ext/detail/GenericSolid.hh#L227)) and then the next from the line in [GenericPlacedVolume](https://github.com/sethrj/celeritas/blob/cms-conversion-fixes/src/celeritas/ext/detail/GenericPlacedVolume.hh#L300).  By-passing those two assertions, I still see the same error message from [the line](https://github.com/sethrj/celeritas/blob/cms-conversion-fixes/src/celeritas/mat/MaterialTrackView.hh#L102).  Please see a test log with demo-geant-integration with cms-run3.gdml, 
[cms-conversion-fixes.log](https://github.com/celeritas-project/celeritas/files/11442728/cms-conversion-fixes.log)

sethrj commented:
It's definitely not a complete fix: we still need @mrguilima to adapt the reflective solid. But the missing materials are definitely part of what needs to be fixed.

But I will add a reflection test to this PR which should let us test that the materials are being mapped correctly.

amandalund commented:
So just to confirm, we should still expect to see this warning with cms-run3?

`GeoMaterialParams.cc:157: warning: Some geometry volumes do not have known material IDs: world_volume@0x7f7110475c60, cms_OCMS@0x7f7110475bc0, cms_CMSE@0x7f7110475b20, cmsextent_CMStoZDC@0x7f7110474c20, cmsextent_ZDCtoFP4201@0x7f7110474cc0, cmsextent_ZDCtoFP4202@0x7f7110474d60, tracker_Tracker@0x7f7110474e00, trackerpatchpannel_TrackerPatchPannel@0x7f7110474400, trackerpatchpannel_PIXSandwichLayer1outer@0x7f711044a060, ...`

sethrj commented:
Hmmm no, I was testing with the Geant integration app. I should try with the straight GDML as well...

sethrj committed:
fixup! Remove labels from temporary bool logical volumes

amandalund commented:
Hmm I was testing with the geant integration app too... and from @whokion's log it looks like he's also still seeing the missing material ID warnings.

sethrj commented:
Thanks @amandalund , I'll continue testing...

sethrj committed:
Merge remote-tracking branch 'upstream/develop' into cms-conversion-fixes

sethrj committed:
Fix pointer-based mapping of reflected volumes

The sensitive detectors are associated with the *underlying* reflected
solid as mapped by the G4ReflectionFactory. This eliminates the warnings
about having to drop the pointer extension to find the correct volume.

sethrj commented:
I'm **definitely not** getting any `Some geometry volumes do not have known material IDs` warnings on this branch running  `demo-geant-integration demo-geant-cpu.mac` in `celeritas-benchmarks/full-cms-test`. The last commit I pushed fixes the mapping of sensitive detector volumes, but I can't replicate the material ID problem...

amandalund commented:
Ahh ok, I was running with `simple-cms-test.mac` (with `cms-run3.gdml` replacing `simple-cms.gdml`). It looks like the difference is yours is set to strip the GDML pointers?

sethrj commented:
Yes, stripping the pointers was necessary (see #713) to get CMS working, and this was before the in-memory conversion (#557) and direct G4LogicalVolume conversion afterward (#739). We should definitely make the pointer stripping on by default if we're using the direct translation layer.

sethrj committed:
Always strip pointers through demo app

We can do that now that in-memory conversion uses the geant pointer
addressess than a mangled name.

amandalund commented:
@sethrj just a couple test failures to fix, but everything looks good!

sethrj committed:
Fix test failure from improved warning message
